### PR TITLE
Donutttts changes rebased

### PR DESCRIPTION
**Pull request by OliverRadini**
Added a player search to the front page, uses player-data-store and data-table, not yet fully functional (not much styling, paging not implemented on front end) but does pull from actual player data.

Worth noting that guild profile player data will no longer work, as the player-data-store now pulls from actual player data, and the actual guild endpoint is not properly implemented yet

### CODE REVIEW

File: herald/src/app/+home/home.component.ts [hunk 1]
```diff
@@ -1,15 +1,51 @@
...
 
   ngOnInit() {
+    this.playerSearchResults = new PlayerDataStore(this.http);
+  }
+
+  /**
+  * does a search for players with the value provided
+  * @param value the player which is being searched for
+  */
+  doPlayerSearch(value: string){
```
Comment (burfo) [inline herald/src/app/+home/home.component.ts]: We shouldn't have API connections all throughout our components.  Shouldn't this be moved to a service class?

File: herald/src/app/shared/player-data/player-data-store.component.ts [hunk 2]
```diff
@@ -158,9 +175,10 @@ export class PlayerDataStore {
     * @returns          an observable which will return the data for the player requested
     */
     getPlayerData(playerName): Observable<SmallPlayerData> {
-      let playerUrl = `/assets/data/players/${playerName}.json`;
+      //let playerUrl = `/assets/data/players/${playerName}.json`;
 
-      return this.http.get(playerUrl)
+      //return this.http.get(playerUrl)//temporary removed whilst testing against real data
+      return this.http.get(`https://uthgard.org/herald/api/player/${playerName}`)
```
Comment (burfo) [inline herald/src/app/shared/player-data/player-data-store.component.ts:181]: We shouldn't have API connections all throughout our components. Shouldn't this be moved to a service class?

**Review by burfo — CHANGES_REQUESTED** [review]

File: herald/src/app/shared/player-data/player-data-store.component.ts (hunk 2 above)
Comment (Donutttt) [inline herald/src/app/shared/player-data/player-data-store.component.ts]: tried building it as a service, the service wouldn't work with http, so I did it this way, the http is being passed in the constructor so it's still DI, just done manually

**Review by burfo — APPROVED** [review]

**Comment by burfo** [comment]
Going to merge this now.  We'll do a separate PR related to a the service and fixing the test error.

**Pull request merged**